### ISSUE TITLE
SC2016 shouldn't trigger for printf -v

### ISSUE THREAD

**Issue by Crestwave** [koalaman/shellcheck#1536]
#### For bugs
- Rule Id (if any, e.g. SC1000): 
- My shellcheck version (`shellcheck --version` or "online"): 
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on shellcheck.net and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [ ] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```bash

#!/usr/bin/env bash
declare -A arr
key=foo
printf -v 'arr[$key]' "bar"
printf '%s\n' "${arr[$key]}"

```

#### Here's what shellcheck currently says:

```

printf -v 'arr[$key]' bar
          ^-- SC2016: Expressions don't expand in single quotes, use double quotes for that.

```

#### Here's what I wanted or expected to see:

Nothing, as in #873 and #1186. I'd actually expect there to be a warning if it *wasn't* single quoted or escaped, as it would then fail on keys like `]`, and it could lead to arbitrary code execution (related: #1322).

**Comment by awgeorge** [comment]
Also doesn't apply using 'mdfind'
